src/write/file.rs

```diff
@@ -86,10 +86,19 @@ impl<W: Write> FileWriter<W> {
         }
     }
 
-    /// Writes the header of the file
-    pub fn start(&mut self) -> Result<()> {
-        self.offset = start_file(&mut self.writer)? as u64;
-        Ok(())
+    /// Writes the header of the file.
+    ///
+    /// This is automatically called by [`Self::write`] if not called following [`Self::new`].
+    ///
+    /// # Errors
+    /// Returns an error if data has been written to the file.
+    fn start(&mut self) -> Result<()> {
```

Comment:
The idea is that users can avoid an extra call `.start`, by having us doing it automatically, right? Nice simplification!

Reply:
Yeah that's right; it's called by write or end, depending on the state